spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ehcache/EhCacheAutoConfiguration.java

```diff
@@ -0,0 +1,68 @@
+package org.springframework.boot.autoconfigure.ehcache;
+
+import net.sf.ehcache.CacheManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cache.ehcache.EhCacheCacheManager;
+import org.springframework.cache.ehcache.EhCacheManagerUtils;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.util.Assert;
+
+import javax.annotation.PostConstruct;
+
+/**
+ * {@link org.springframework.boot.autoconfigure.EnableAutoConfiguration Auto-configuration} for EhCache.
+ *
+ * @author Eddú Meléndez
+ * @since 1.3.0
+ */
+@Configuration
+@ConditionalOnClass(CacheManager.class)
+public class EhCacheAutoConfiguration {
```

Comment:
It's better but we're not there yet. I think the trigger should also be the presence of a Spring `CacheManager`. If one is configured we should just back-off